scripts/prepare_disk.sh

```diff
@@ -9,16 +9,54 @@
 set -eu
 
 PARTITION_TABLE_FILE="${PARTITION_TABLE_FILE:-}"
+TARGET_DISK="${TARGET_DISK:-}"
+
+BOOT_PARTITION_NAME="boot"
+BOOT_PARTITION_START="2048"
 
 usage()
 {
-    cat <<-EOT
-	Usage: ${0} [OPTIONS] <DISK>
-	Prepare the target DISK to a predefined disk layout.
-	  -t Partition table file (mandatory).
-	  -h Print this help text and exit
-	NOTE: This script is destructive and will destroy your data.
-EOT
+    echo "Usage: ${0} [OPTIONS] <DISK>"
+    echo "Prepare the target DISK to a predefined disk layout."
+    echo "  -t Partition table file (mandatory)."
+    echo "  -h Print this help text and exit"
+    echo "NOTE: This script is destructive and will destroy your data."
+}
+
+partition_resize()
+{
+    if ! sha512sum -csw "${PARTITION_TABLE_FILE}.sha512"; then
+        echo "Error processing partition table: crc error."
+        exit 1
+    fi
+
+    while IFS="${IFS}:=," read -r label _ start _ size _ _ _; do
+        # Invalidate lines that are: empty, start with #, start and size are
+        # not integers by using integer comparison test.
+        if [ -z "${label}" ] || \
+           [ "${label%%"${label##?}"}" = "#" ] || \
+           ! [ "${start}" -eq "${start}" ] 2> /dev/null || \
+           ! [ "${size}" -eq "${size}" ] 2> /dev/null; then
+            continue
+        fi
+
+        if [ "${label}" = "${BOOT_PARTITION_NAME}" ] && \
```

Comment:
The label part from the partition table cannot be used as filesystem label, the partition table is not aware of filesystem types, and this information is not dumped by fdisk. 
It makes it hard to test and it is also not trivial when you want to add more storage layouts later